Lib/typing.py

```diff
@@ -1404,10 +1404,43 @@ def _get_protocol_attrs(cls):
                 attrs.add(attr)
     return attrs
 
+_CLASSVAR_PREFIXES = ("typing.ClassVar", "t.ClassVar", "ClassVar")
 
-def _is_callable_members_only(cls):
+def _is_callable_or_classvar_members_only(cls, instance):
+    """Returns a 2-tuple signalling two things:
+    (Valid protocol?, If not valid protocol, was it due to ClassVar value mismatch?)
+    """
+    attr_names = _get_protocol_attrs(cls)
+    annotations = getattr(cls, '__annotations__', {})
     # PEP 544 prohibits using issubclass() with protocols that have non-method members.
-    return all(callable(getattr(cls, attr, None)) for attr in _get_protocol_attrs(cls))
+    # bpo-44975: Relaxing that restriction to allow for runtime-checkable
+    # protocols with class variables since those should be available at class
+    # definition time.
+    for attr_name in attr_names:
+        attr = getattr(cls, attr_name, None)
+        # Method-like.
+        if callable(attr):
+            continue
+        annotation = annotations.get(attr_name)
+        # ClassVar member
+        if getattr(annotation, '__origin__', None) is ClassVar:
+            instance_attr = getattr(instance, attr_name, None)
+            # If we couldn't find anything, don't bother checking value types.
+            if (instance_attr is not None
+                and attr is not None
+                and type(instance_attr) != type(attr)):
+                return False, True
+            continue
+        # ClassVar string annotations (forward references).
+        if isinstance(annotation, str) and annotation.startswith(_CLASSVAR_PREFIXES):
+            instance_attr = getattr(instance, attr_name, None)
+            if (instance_attr is not None
+                and attr is not None
+                and type(instance_attr) != type(attr)):
```

Comment:
This is incorrect. Consider this example:

```python
class P(Protocol):
     x: ClassVar[object] = object()

class Y:
     x: ClassVar[object] = 1
```
Y implements protocol P, but your code rejects it. There are also problems with None here that could produce an incorrect result.

I'd recommend removing this whole check, and just checking that the attribute exists. Full runtime type checking is hard and the standard library shouldn't try to do it.